src/torchrunx/utils.py

```diff
@@ -21,6 +25,23 @@ def get_open_port() -> int:
     return port
 
 
+def automatic() -> tuple[list[str], int]:
```

Comment:
Maybe split this into `auto_hosts` and `auto_workers`?

Reply:
additionally, let's put it in `auto.py`

Reply:
or a more general `environment.py` and then we put all the `auto` and `slurm` functions in there?